### PR TITLE
APP-22: Address update card

### PR DESCRIPTION
**Pull request by hugokallstrom**
This is shown in the insurance details view when an upcoming agreement table is found for one of the contracts. Design is found here: https://www.figma.com/file/Ds2hXzDbnUt9ZNhg4HSWH3/Moving-Flow-Android?node-id=118%3A5198

device-2021-05-10-111722
device-2021-05-10-111711
device-2021-05-10-111653

Should we change the color of `ThemeOverlay.Hedvig.InfoCard` to better reflect the design?

### Checklist

- [x] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/res/layout/your_info_home.xml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (hedvigoscar) [inline app/src/main/res/layout/your_info_home.xml:2]: Couldn't this be just a single `ConstrantLayout`? Or a set of nested `LinearLayout`s? 🤔

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractExt.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package com.hedvig.app.feature.insurance.ui.detail
+
+import com.hedvig.android.owldroid.graphql.InsuranceQuery
+import com.hedvig.app.R
+import com.hedvig.app.feature.insurance.ui.detail.yourinfo.YourInfoModel
+import com.hedvig.app.util.apollo.stringRes
+
+fun InsuranceQuery.Contract.toModelItems(): List<YourInfoModel> = when {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractExt.kt:8]: How come we switched from the Row-based architecture to the more compounded structure here? 😄

**Review by hedvigoscar — COMMENTED** [review]
A lot of code to digest 😄

**Comment by hedvigoscar** [comment]
Regarding the info card theme, yes, that would be appropriate I think 🙂

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/CoverageFragment.kt [hunk 3]
```diff
@@ -19,43 +19,27 @@ class CoverageFragment : Fragment(R.layout.contract_detail_coverage_fragment) {
     private val binding by viewBinding(ContractDetailCoverageFragmentBinding::bind)
     private val model: ContractDetailViewModel by sharedViewModel()
     private val requestBuilder: RequestBuilder<PictureDrawable> by inject()
+    private val coverageAdapter: CoverageAdapter by lazy {
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/CoverageFragment.kt:22]: Good idea!

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailViewModel.kt [hunk 4]
```diff
@@ -5,51 +5,105 @@ import androidx.lifecycle.MutableLiveData
...
+        }
+    }
 
-            if (response.getOrNull()?.hasErrors() == true) {
-                _data.postValue(Result.failure(Error()))
-                return@launch
-            }
+    private suspend fun createContractItems(contract: InsuranceQuery.Contract): List<YourInfoModel> {
+        val contractItems = contract.toModelItems()
+        val upcomingAgreementItem = getUpcomingAgreementItem()
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailViewModel.kt]: I moved the upcoming agreement query to a fragment and reused it in the `insurance.graphql`-query.

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractExt.kt (hunk 2 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractExt.kt:8]: I think its difficult to find the right abstraction level for recyclerviews with different view types. For `YourInfoModel.Home` we should even use the new contract extension field, `currentAgreementDetailsTable()` (i'll create another PR for that). So I suppose we need to use a more Row-based architecture then :)

**Review by hedvigoscar — APPROVED** [review]

**Pull request merged**